homeassistant/components/sensor/sensehat.py

```diff
@@ -0,0 +1,139 @@
+"""
+Support for Sense HAT sensors.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/
+"""
+import os
+import logging
+from homeassistant.const import TEMP_CELSIUS
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['https://github.com/RPi-Distro/python-sense-hat'
+                '/archive/master.zip#sense-hat==2.2.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_cpu_temp():
+    """Get CPU temperature."""
+    res = os.popen("vcgencmd measure_temp").readline()
+    t_cpu = float(res.replace("temp=", "").replace("'C\n", ""))
+    return t_cpu
+
+
+def get_average(temp_base):
+    """Use moving average to get better readings."""
+    if not hasattr(get_average, "temp"):
+        get_average.temp = [temp_base, temp_base, temp_base]
+    get_average.temp[2] = get_average.temp[1]
+    get_average.temp[1] = get_average.temp[0]
+    get_average.temp[0] = temp_base
+    temp_avg = (get_average.temp[0]+get_average.temp[1]+get_average.temp[2])/3
+    return temp_avg
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the sensor platform."""
+    add_devices([AddTemperature(hass)])
+    add_devices([AddHumidity(hass)])
+    add_devices([AddPressure(hass)])
+
+
+class AddTemperature(Entity):
```

Comment:
`class AddTemperature(Entity):`, `class AddHumidity(Entity):`, and `class AddPressure(Entity):` are very similar. Take a look at other platforms like `systemmonitor`, `bitcoin`, or `yweather` to use a `SENSOR_TYPES` dict to create multiple type of sensors and an object that holds the data.

I would like also to suggest that you use a throttle.